### PR TITLE
1166 flexible csv separator selector

### LINKED ISSUE #1166: CsvSeparatorSelector from R
There should be an implementation of ICsvSeparatorSelector for R where the selector could be changed.

### PR DESCRIPTION
**Pull request by abdelr**
Fixes #1166 (Flexible csv separator selector)

### CODE REVIEW

File: tests/OSPSuite.R.Tests/Services/DataImporterSpecs.cs [hunk 1]
```diff
@@ -51,19 +21,19 @@ public class When_importing_data_from_r : concern_for_DataImporter
       [Observation]
       public void should_import_simple_data()
       {
-         var _importerConfiguration = getConfiguration("importerConfiguration1.xml");
-         var dataRepositories = sut.ImportFromConfiguration(_importerConfiguration, _metaDataCategories, _columnInfos, _dataImporterSettings, getFileFulName("sample1.xlsx"));
+         sut.ImportXslxFromConfiguration(getFileFulName("importerConfiguration1.xml"), getFileFulName("sample1.xlsx")).Count.ShouldBeEqualTo(1);
```
Comment (Yuri05) [inline tests/OSPSuite.R.Tests/Services/DataImporterSpecs.cs:24]: yeah, that is finally what I expected from the very beginning :)
There are more use cases for sure, but this one (importing a data file based on existing configuration) is the most important for me.

**Review by Yuri05 — APPROVED** [review]

File: tests/OSPSuite.R.Tests/Services/DataImporterSpecs.cs [hunk 2]
```diff
@@ -1,46 +1,16 @@
 ﻿using OSPSuite.BDDHelper;
 using OSPSuite.BDDHelper.Extensions;
-using OSPSuite.Core.Serialization;
```
Comment (Yuri05) [inline tests/OSPSuite.R.Tests/Services/DataImporterSpecs.cs:3]: let's rename the file to _DataImporter**Task**Specs.cs_

File: tests/OSPSuite.R.Tests/Services/DataImporterTaskSpecs.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+{
+   public abstract class concern_for_DataImporter : ContextForIntegration<IDataImporterTask>
+   {
+      protected override void Context()
+      {
+         base.Context();
+         sut = Api.GetDataImporterTask();
+      }
+
+      protected string getFileFulName(string fileName) => Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "Data", fileName);
```
Comment (Yuri05) [inline tests/OSPSuite.R.Tests/Services/DataImporterTaskSpecs.cs:16]: typo: `getFileFulName` (sorry, didn't see it last time)

Comment (abdelr) [reply tests/OSPSuite.R.Tests/Services/DataImporterTaskSpecs.cs]: Well, better later than never, right? ;) I did not see it before either and I wrote the code myself...

**Review by Yuri05 — APPROVED** [review]

**Pull request merged**